src/base/memory_monitor_sr.cpp

```diff
@@ -202,38 +206,71 @@ namespace cubperf
 
   void mmon_subcomponent::add_cur_stat (uint64_t size)
   {
-    m_cur_stat.fetch_add (size);
+    m_cur_stat += size;
   }
 
   void mmon_subcomponent::sub_cur_stat (uint64_t size)
   {
-    m_cur_stat.fetch_sub (size);
+    m_cur_stat -= size;
   }
 
   const char *mmon_component::get_name ()
   {
     return m_comp_name.c_str ();
   }
 
-  void mmon_component::add_stat (uint64_t size, int subcomp_idx, bool init, bool expand)
+  void mmon_component::add_stat (uint64_t size, int subcomp_idx, bool allocation_at_init)
   {
     /* description of add_stat(..).
-     * 1) if init == true, add init_stat
+     * 1) if allocation_at_init == true, add init_stat
      * 2) then, add cur_stat
      * 3) compare with peak_stat
      * 4) if cur_stat(new) > peak_stat, update peak_stat
-     * 5) if expand == true, expand_count++
-     * 6) call subcomponent->add_cur_stat(size) */
-    return;
+     * 5) call subcomponent->add_cur_stat(size) */
+
+    if (allocation_at_init)
+      {
+	m_stat.init_stat += size;
+      }
+
+    m_stat.cur_stat += size;
+
+    if (m_stat.cur_stat > m_stat.peak_stat)
+      {
+	m_stat.peak_stat = m_stat.cur_stat.load ();
```

Comment:
Nit. As fas as I know, it's the same.
```suggestion
	m_stat.peak_stat = m_stat.cur_stat;
```

Reply:
It makes error because of atomic::atomic(atomic &)

Reply:
oh, I see.